accord-core/src/test/java/accord/impl/basic/Cluster.java

```diff
@@ -301,4 +345,147 @@ public static EnumMap<MessageType, Stats> run(Id[] nodes, Supplier<PendingQueue>
             lookup.values().forEach(Node::shutdown);
         }
     }
+
+    private static BiFunction<Id, Id, Link> partition(List<Id> nodes, RandomSource random, int rf, BiFunction<Id, Id, Link> defaultLinks)
+    {
+        Collections.shuffle(nodes, random.asJdkRandom());
```

Comment:
```suggestion
        List<Id> nodes = new ArrayList<>(nodes);
        Collections.shuffle(nodes, random.asJdkRandom());
```

mutating your input can be a source of bugs; best to treat it as immutable

Reply:
We take a copy of the list externally to the generators, and invoke them in a single thread. So I don't think this is necessary, but if you strongly prefer it I don't mind applying a local copy as well since this is not performance critical code.

Reply:
ll leave this as a +0... tend to find coupling like this change over time and cause subtle issues so I prefer to be more defensive, but if you know the ownership within the whole context is correct, we can leave as is.